pkg/analysis/passes/modulejs/modulejs.go

```diff
@@ -24,7 +23,7 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	archiveDir, ok := pass.ResultOf[archive.Analyzer].(string)
 	if !ok || archiveDir == "" {
 		// this should never happen
-		return nil, fmt.Errorf("archive dir not found")
+		return nil, nil
```

Comment:
Intended. The error should come from the archive validator, this is redundant logic,. we can skip this validator if there's no archive